### PR TITLE
Feat: Add `insert_many` method

### PR DESCRIPTION
**Pull request by ischaojie**
Support class method `insert_many` and `insert_one` https://github.com/aminalaee/mongox/issues/46

Refer to motor [insert_many](https://motor.readthedocs.io/en/stable/api-tornado/motor_collection.html#motor.motor_tornado.MotorCollection.insert_many) document.

### CODE REVIEW

File: mongox/models.py [hunk 1]
```diff
@@ -336,6 +336,23 @@ async def insert(self: T) -> T:
         self.id = result.inserted_id
         return self
 
+    @classmethod
+    async def insert_one(cls: typing.Type[T], model: T) -> T:
```
Comment (aminalaee) [inline mongox/models.py:340]: To be honest I'm not a huge fan of this one. The final result will be:

```python
movie = Movie()
await Movie.insert_one(movie)
```

Though this is simpler:

```python
await Movie().insert()

# Or
movie = Movie()
await movie.insert()
```

Is there any reason for that?

File: tests/test_models.py [hunk 2]
```diff
@@ -492,3 +493,51 @@ async def test_model_get_by_id() -> None:
...
+
+    with pytest.raises(TypeError):
+        book = Book(name="The Book", year=1972)
+        await Movie.insert_one(book)  # type: ignore
+
+
+async def test_model_insert_many() -> None:
+    movies = []
+    movie_names = ["The Dark Knight", "The Dark Knight Rises", "The Godfather"]
+    for movie_name in movie_names:
```
Comment (aminalaee) [inline tests/test_models.py:528]: Can we do list comprehension here?

File: mongox/models.py [hunk 3]
```diff
@@ -336,6 +336,23 @@ async def insert(self: T) -> T:
...
+        if not isinstance(model, cls):
+            raise TypeError(f"Model must be of type {cls.__name__}")
+        return await model.insert()
+
+    @classmethod
+    async def insert_many(cls: typing.Type[T], models: typing.List[T]) -> typing.List[T]:
+        if not all(isinstance(model, cls) for model in models):
+            raise TypeError(f"All models must be of type {cls.__name__}")
+
+        datas = (model.dict(exclude={"id"}) for model in models)
```
Comment (aminalaee) [inline mongox/models.py:350]: Maybe `data` instead of `datas`?

**Review by aminalaee — COMMENTED** [review]
This is great, thank you for the PR.

File: mongox/models.py (hunk 1 above)
Comment (ischaojie) [reply mongox/models.py:340]: `instance.insert()`	provide a convenient way to insert data, use `insert_one` just want to keep behavior consistent with `insert_many()`,  they all belong to the class method.

Maybe use `instance.insert` is enough.

Comment (ischaojie) [reply mongox/models.py:340]: Maybe we should provide `Movie.insert(movie)` method 😁

Comment (aminalaee) [reply mongox/models.py:340]: I don't know. I would try to keep this as small as possible. Having to deal with multiple ways of doing the same thing can be challenging 😄

**Comment by ischaojie** [comment]
I removed `insert_one` method, maybe provide it at the appropriate time.

File: mongox/models.py (hunk 1 above)
Comment (aminalaee) [reply mongox/models.py]: By the way, this needs to be documented too. Looks great now.

**Review by aminalaee — APPROVED** [review]
Thanks a lot for the PR 🚀

**Pull request merged**